lib/phoenix/socket/transport.ex

```diff
@@ -258,11 +258,13 @@ defmodule Phoenix.Socket.Transport do
     [connect_info: connect_info] ++ config
   end
 
+  # The original session_config is returned in addition to init value so we can
+  # access special config like :csrf_token_key downstream.
   defp init_session(session_config) when is_list(session_config) do
     key = Keyword.fetch!(session_config, :key)
     store = Plug.Session.Store.get(Keyword.fetch!(session_config, :store))
     init = store.init(Keyword.drop(session_config, [:store, :key]))
-    {key, store, init}
+    {key, store, {session_config, init}}
```

Comment:
We only need the `csrf_token_key`, so what if we only pass it here?

```suggestion
    csrf_token_key = Keyword.get(session_config, :csrf_token_key, "_csrf_token")
    {key, store, {csrf_token_key, init}}
```